gdal/port/cpl_odbc.cpp

```diff
@@ -112,10 +116,32 @@ int CPLODBCDriverInstaller::InstallDriver( const char* pszDriver,
         }
 
         // Try to install ODBC driver in new location.
-        if( FALSE == SQLInstallDriverEx(pszDriver, nullptr, m_szPathOut,
+        if( FALSE == SQLInstallDriverEx(pszDriver, pszEnvHome, m_szPathOut,
                                         ODBC_FILENAME_MAX, nullptr, fRequest,
                                         &m_nUsageCount) )
         {
+            // if installing the driver fails, we need to roll back the changes to ODBCSYSINI environment
+            // variable or all subsequent use of ODBC calls will fail
+            char * pszEnvRollback = nullptr;
+            if ( pszEnvOdbcSysIni )
+            {
+                const size_t nLen = strlen( pszEnvOdbcSysIni ) + 12;
+                pszEnvRollback = static_cast<char *>(CPLMalloc(nLen));
+                snprintf( pszEnvRollback, nLen, "ODBCSYSINI=%s", pszEnvOdbcSysIni );
+            }
+            else
+            {
+                // ODBCSYSINI not previously set, so remove
+                const size_t nLen = 11;
+                pszEnvRollback = static_cast<char *>(CPLMalloc(nLen));
+                snprintf( pszEnvRollback, nLen, "ODBCSYSINI" );
```

Comment:
https://man7.org/linux/man-pages/man3/putenv.3.html
> The GNU C library implementation provides a nonstandard extension. If string does not include an equal sign:
>            `putenv("NAME");`
> then the named variable is removed from the caller's environment.

So, this probably is fine for Unix.

But, there needs to be `ODBCSYSINI=` for MSVC, I think, as per earlier comment https://github.com/OSGeo/gdal/pull/2838#discussion_r466874285

Reply:
you could also more simply use ```pszEnvRollback = CPLStrdup("ODBCSYSINI=");```

Reply:
Thanks -- how's the new commit look?

Reply:
LGTM, Thx!